src/uchen/tboard/tboard_file.cc

```diff
@@ -1,25 +1,167 @@
 #include "uchen/tboard/tboard_file.h"
 
+#include <climits>
+#include <filesystem>
 #include <fstream>
 #include <ios>
+#include <sstream>
+#include <string>
+
+#ifdef _WIN32
+#include <winsock2.h>
+#endif
 
 #include "absl/log/log.h"
 
+#include "crc32c/crc32c.h"
+#include "proto/event.pb.h"
+#include "proto/summary.pb.h"
 #include "src/uchen/tboard/tboard_file.h"
 
+namespace {
+
+// MaskDelta is a fixed arbitrary number that is used to mask the CRC32
+// checksum. expected from TensorBoard.
+constexpr uint32_t kMaskDelta = 0xa282ead8ul;
+
+}
+
 namespace uchen::tboard {
 
-absl::StatusOr<TBoardFile> TBoardFile::Open(std::string_view path) {
-  std::ofstream file(std::string(path), std::ios::binary);
+absl::StatusOr<TBoardFile> TBoardFile::Open(std::string_view logs_directory,
+                                            std::string_view log_name) {
+  if (logs_directory.empty()) {
+    return absl::InvalidArgumentError("Log file name must be provided");
+  }
+
+  std::string timestamp =
+      absl::FormatTime("%Y%m%d-%H%M%S", absl::Now(), absl::UTCTimeZone());
+  std::filesystem::path events_directory =
+      std::filesystem::path(logs_directory) / std::filesystem::path(timestamp);
```

Comment:
This directory should come from the outside (e.g. as a function parameter). This is a library so it should be as idempotent  as possible - e.g. calling the API with same arguments should yield the same result. E.g. rerunning same training (say, because we crashed) should be able to override/add to docs.